### PR TITLE
docs: describe where the build system is evolving to

### PR DESCRIPTION
**Pull request by compnerd**
Add a manifesto which describes where the build infrastructure is slowly
evolving to.  Although a good chunk of the necessary work to enable this
is already in the tree, this helps explain the desired state to everyone
working on the project.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @gottesmm - hopefully this covers what you wanted

File: docs/BuildManifesto.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  /* stdlib */   (build: "x86_64-apple-macosx10.14", host: "armv7-apple-ios12.3"),
+]
+```
+
+This simplifies the build terminology by having to only deal with the terms
+`build` and `host` which removes some of the confusion caused by the original
+build system's implementation creating multiple terms and being designed around
+the needs of the Darwin build.
+
+This also generalises to allow for multiple cross-compilations in a single shot
```
Comment (gottesmm) [inline docs/BuildManifesto.md]: single shot => single build invocation

File: docs/BuildManifesto.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+which allows the functionality currently available only on Darwin to be applied
+to all the targets that Swift supports (and may support in the future).
+
+Doing so also enables the build system to be trimmed significantly as we can use
+CMake as it was designed to while retaining the ability to cross-compile (which
```
Comment (gottesmm) [inline docs/BuildManifesto.md]: Elaborate?

File: docs/BuildManifesto.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+## Current State
+
+Currently, the Swift build is setup similar to the way that the GCC build was
+setup.  This requires the understanding of the terms `build`, `host`, `target`
+as used by autotools.
+
+<dl>
+  <dt><strong>build</strong></dt>
+  <dd>the system where the package is being configured and compiled</dd>
```
[the remark this answers is not in the document]
Comment (compnerd) [reply docs/BuildManifesto.md]: I've done the same using Linux -> Windows -> android

[the remark this answers is not in the document]
Comment (gottesmm) [reply docs/BuildManifesto.md]: Sure. I want an example in the document.

[the remark this answers is not in the document]
Comment (gottesmm) [reply docs/BuildManifesto.md]: I think that is fine. I would also put in a normal macOS one as well for iOS.

**Comment by CodaFi** [comment]
@compnerd Is this ready to be merged?

**Comment by compnerd** [comment]
@CodaFi I believe so - I wanted to make sure that all the points that @gottesmm wanted noted have been

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**